pkg/forwarder/forwarder_health.go

```diff
@@ -111,20 +112,15 @@ func (fh *forwarderHealth) healthCheckLoop() {
 	}
 }
 
-// computeDomainsURL populates a map containing API Endpoints per API keys that belongs to the forwarderHealth struct
-func (fh *forwarderHealth) computeDomainsURL() {
-	for domain, apiKeys := range fh.keysPerDomains {
-		apiDomain := ""
-		re := regexp.MustCompile("datadoghq.[a-z]*")
-		if re.MatchString(domain) {
-			apiDomain = "https://api." + re.FindString(domain)
-		} else {
-			apiDomain = domain
+// computeKeysPerAPIDomain populates a map containing API Endpoints per API keys that belongs to the forwarderHealth struct
+func (fh *forwarderHealth) computeKeysPerAPIDomain(keysPerDomains map[string][]string) {
+	fh.keysPerAPIDomain = make(map[string][]string)
+	for domain, apiKeys := range keysPerDomains {
+		if apiDomain, err := config.ComputeAPIDomain(domain); err == nil {
```

Comment:
if there is an error, I think we'll want to log something here with a message that includes the error returned